### PR TITLE
defaultValues with field arrays and yup stopped working in 6.3.1 #3342

### PR DESCRIPTION
**Pull request by mortenk**

### CODE REVIEW

File: src/controller.test.tsx [hunk 1]
```diff
@@ -772,7 +772,7 @@ describe('Controller', () => {
       const Component = () => {
         const { control } = useForm({
           defaultValues: {
-            test: [{ data: '' }],
+            test: [{}],
```
Comment (bluebill1049) [inline src/controller.test.tsx:775]: why is this changed?

File: src/useFieldArray.ts [hunk 2]
```diff
@@ -482,6 +480,7 @@ export const useFieldArray = <
   React.useEffect(() => {
     const resetFunctions = resetFieldArrayFunctionRef.current;
     const fieldArrayNames = fieldArrayNamesRef.current;
+    fieldArrayNames.add(name);
```
Comment (bluebill1049) [inline src/useFieldArray.ts:483]: nice.

File: src/useFieldArray.test.tsx [hunk 3]
```diff
@@ -3708,6 +3708,38 @@ describe('useFieldArray', () => {
       ]);
     });
 
+    it('should populate all fields with default values', () => {
```
Comment (bluebill1049) [inline src/useFieldArray.test.tsx:3711]: Thanks for the test.

**Review by bluebill1049 — COMMENTED** [review]
Thanks for the PR 👍 and your effort. I think we will still need to use defaultValue, because all the cypres testing still using defaultVlaue and it's going to raise issues when you actually remove all the default values (which combine the usage with all the other actions from useFieldArray). Happy to merge this PR tho.

File: src/controller.test.tsx (hunk 1 above)
Comment (mortenk) [reply src/controller.test.tsx:775]: The test verifies that a warning is printed to the console if defaultValue is missing from either the input or useForm(), but the my fix makes defaultValue no longer required for the input field. The value is populated from the defaultValues passed to useForm() and no warning is printed to the console anymore.

Comment (bluebill1049) [reply src/controller.test.tsx:775]: It works with default populate, but not when you manipulate the fields (those actions). I will merge the PR anyhow, thanks for your effort.
